modules/table/src/test/java/org/apache/ignite/internal/table/distributed/TableManagerTest.java

```diff
@@ -245,37 +222,36 @@ public class TableManagerTest extends IgniteAbstractTest {
     /** Hybrid clock. */
     private final HybridClock clock = new HybridClockImpl();
 
+    /** Catalog vault. */
+    private VaultManager catalogVault;
```

Comment:
Catalog doesn't have own Value, and no Vault implementation is backed by Catalog.
```suggestion
    private VaultManager vaultManager;
```

Reply:
In the current test, the catalog has its own valt and metastore, the one used in the test is mocked.

Then the question may arise, they say, why are they needed at all? 
We need a `Consumer<LongFunction<CompletableFuture<?>>> registry` (for vv entities inside **TableManager**) that used to be from the configuration, but now it needs to be provided somehow. If there is a more elegant way to do this, then welcome.

Reply:
Since there is no written answer (only emoji), I assume that Andrey agrees with my comment.

Reply:
I think you can pass a CatalogEvent directly to TableManager listeners.

And I don't understand the purpose of `mockManagersAndCreateTableWithDelay` test.
If the test check interactions between CatalogManager and TableManager, then it should be proper integration test.
If it just a unit, that checks guarantee the table will be properly initialized, then passing an event is enough.

Reply:
First of all, why do you guys need vault here? MS doesn't use it to store key projections any longer. Seems that it's only used for persisting applied revision which is also effectively deprecated.

Reply:
> I think you can pass a CatalogEvent directly to TableManager listeners.

I don’t understand how this can help, I need exactly:
`revisionUpdater = (LongFunction<CompletableFuture<?>> function) -> catalogMetastore.registerRevisionUpdateListener(function::apply);`

> First of all, why do you guys need vault here? MS doesn't use it to store key projections any longer. Seems that it's only used for persisting applied revision which is also effectively deprecated.

I just want the tests to work when changing to a directory, then we can redo that later.

Reply:
Fine to me